graphql/schemabuilder/function.go

```diff
@@ -238,7 +238,7 @@ func (funcCtx *funcContext) argsTypeMap(argType graphql.Type) (map[string]graphq
 
 // prepareResolveArgs converts the provided source, args and context into the
 // required list of reflect.Value types that the function needs to be called.
-func (funcCtx *funcContext) prepareResolveArgs(source interface{}, args interface{}, ctx context.Context) []reflect.Value {
+func (funcCtx *funcContext) prepareResolveArgs(source interface{}, hasArgs bool, args interface{}, ctx context.Context, selectionSet *graphql.SelectionSet) []reflect.Value {
```

Comment:
confused about what should be an argument and what should live on funcCtx

Reply:
Ideally, anything that doesn't change should be stored on the funcContext.  Things that change should be passed in.  Might be worth wrapping back around to this.